scalafix-core/shared/src/main/scala/scalafix/internal/rule/DiscardedValues.scala

```diff
@@ -0,0 +1,80 @@
+package scalafix.internal.rule
+
+import scala.meta.Term.Block
+import scala.meta._
+import scalafix.SemanticdbIndex
+import scalafix.internal.util.TypeSyntax
+import scalafix.lint.{LintCategory, LintMessage}
+import scalafix.rule.{RuleCtx, RuleName, SemanticRule}
+import scalafix.syntax._
+import scala.meta.contrib.implicits.Equality._
+
+case class DiscardedValues(index: SemanticdbIndex)
+    extends SemanticRule(index, RuleName("DiscardedValues")) {
+
+  override def description: String =
+    "This rule ensures that statements should only return Unit"
+
+  private lazy val errorCategory: LintCategory =
+    LintCategory.error("Statements should only return Unit")
+
+  def litType(l: Lit): Option[Type] = l match {
+    case Lit.Int(_) => Some(t"_root_.scala.Int")
+    case Lit.Unit() => Some(t"_root_.scala.Unit")
+    case Lit.Float(_) => Some(t"_root_.scala.Float")
+    case Lit.Double(_) => Some(t"_root_.scala.Double")
+    case Lit.Symbol(_) => Some(t"_root_.scala.Symbol")
+    case Lit.Boolean(_) => Some(t"_root_.scala.Boolean")
+    case Lit.Byte(_) => Some(t"_root_.scala.Byte")
+    case Lit.Char(_) => Some(t"_root_.scala.Char")
+    case Lit.Long(_) => Some(t"_root_.scala.Long")
+    case Lit.Null() => Some(t"_root_.scala.Null")
+    case Lit.String(_) => Some(t"_root_.scala.String")
+    case Lit.Short(_) => Some(t"_root_.scala.Short")
+    case _ => None
+  }
+
+  def statementType(stat: Stat): Option[Type] = stat match {
+    case l: Lit => litType(l)
+    case name: Term.Name => name.symbol.flatMap(_.resultType)
+    case Term.Select(_, name) => statementType(name)
+    case Term.Apply(fun, _) => statementType(fun)
+    case Term.ApplyInfix(_, op, _, _) => statementType(op)
+    case Term.ApplyUnary(op, _) => statementType(op)
+    case _ => None
+  }
+
+  override def check(ctx: RuleCtx): List[LintMessage] = {
+    def statsToErrors(stats: List[Stat]): List[LintMessage] =
+      for {
+        stat <- stats
+        statType <- statementType(stat)
```

Comment:
![image](https://user-images.githubusercontent.com/4629312/34959249-c0a3a8f0-fa46-11e7-8ab3-340dd5e17f89.png)

Reply:
Yeah, I'm afraid we'll have to wait a bit longer for this to happen. The blocking issue is https://github.com/scalameta/scalameta/issues/1212 I'm gonna close this issue to clear the PR queue.